vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java

```diff
@@ -305,8 +306,13 @@ public void setDataProvider(DataProvider<T, ?> dataProvider) {
                 .withProperty("name",
                         value -> String.valueOf(valueProvider.apply(value))));
         column.setComparator(
-                ((a, b) -> compareMaybeComparables(valueProvider.apply(a),
-                        valueProvider.apply(b))));
+                new SerializableComparator<T>() {
```

Comment:
Would setting the comparator to a variable work? It would seem simpler to me, and also maybe keep sonarqube happy.
```java
        SerializableComparator<T> comparator = (a,
                b) -> compareMaybeComparables(valueProvider.apply(a),
                        valueProvider.apply(b));
        column.setComparator(comparator);
 ```

Reply:
Could you add a unit test that reproduces the error?

Reply:
> Would setting the comparator to a variable work?
Nope, it wont work. The problem would the same as now.

Reply:
> Could you add a unit test that reproduces the error?
I am not familiar enough with project structure, so I leave this to your team. Guillermo told, that you actually should have already some serialization tests, but for some reason they did not cover this, or maybe skipped.

Reply:
I will merge this to a new branch I created with the test.